File: app/src/main/java/closer/vlllage/com/closer/pool/PoolApplication.java
package closer.vlllage.com.closer.pool;

import android.app.Application;

public abstract class PoolApplication extends Application {
    private final Pool pool = new Pool();

    @Override
    public void onTerminate() {
        pool.end();
        super.onTerminate();
    }

    protected <T extends PoolMember> T $(Class<T> member) {
        return pool.$(member);
    }

    public Pool getPool() {
        return pool;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/pool/PoolApplication.kt
package closer.vlllage.com.closer.pool

import androidx.multidex.MultiDexApplication
import com.queatz.on.On

abstract class PoolApplication : MultiDexApplication() {

    internal val on = On()

    override fun onTerminate() {
        on.off()
        super.onTerminate()
    }
}
